src/test/java/com/example/bookstore/controller/CategoryControllerIntegrationTest.java

```diff
@@ -0,0 +1,195 @@
+        CategoryDto expected = new CategoryDto()
+                .setId(1L)
+                .setName("Detective");
+        MvcResult result = mockMvc.perform(post("/categories")
+                        .content(requestJson)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+        CategoryDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), CategoryDto.class);
+        assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    void createCategory_InValidData_BadRequest() throws Exception {
+        CategoryDto invalidCategoryRequest = new CategoryDto();
+        String requestJson = objectMapper.writeValueAsString(invalidCategoryRequest);
+        mockMvc.perform(post("/categories")
+                        .content(requestJson)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Find and return list of existing categories")
+    @Sql(scripts = "classpath:database/add-for-category-tests.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    void getAll_ValidRequest_ReturnListOfCategories() throws Exception {
+        MvcResult result = mockMvc.perform(get("/categories"))
+                .andExpect(status().isOk())
+                .andReturn();
+        List<CategoryDto> expected = new ArrayList<>();
+        expected.add(new CategoryDto().setId(1L).setName("Fantasy"));
+        CategoryDto[] actual = objectMapper.readValue(result
+                .getResponse().getContentAsString(), CategoryDto[].class);
+        assertEquals(expected, Arrays.stream(actual).toList());
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Get category by id")
+    @Sql(scripts = "classpath:database/add-for-category-tests.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    void getById_ValidId_ReturnCategory() throws Exception {
+        MvcResult result = mockMvc.perform(get("/categories/1")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+        CategoryDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), CategoryDto.class);
+        assertNotNull(actual);
+        assertEquals(actual.getId(), 1L);
+        assertEquals(actual.getName(), "Fantasy");
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @DisplayName("Update existing category by Id")
+    @Sql(scripts = "classpath:database/add-for-category-tests.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    public void updateCategory_validId_Success() throws Exception {
```

Comment:
remove it everywhere or add it for all test cases
```suggestion
    void updateCategory_validId_Success() throws Exception {
```